### PR TITLE
[BUG] fixing some timeout weirdness

### PR DESCRIPTION
**Pull request by rksh**

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java [hunk 1]
```diff
@@ -313,8 +315,13 @@ public void getApiToken(@NotNull String username, @NotNull String password, @Not
                         completedListener.onTaskFailed(response.code(), null);
                     } else {
                         if (null != responseBody) {
-                            completedListener.onTaskSucceeded(new Gson().fromJson(responseBody.charStream(),
+                            try {
+                                completedListener.onTaskSucceeded(new Gson().fromJson(responseBody.charStream(),
                                     ApiTokenResponse.class));
+                            } catch (JsonSyntaxException e) {
+                                //ALIBI: sometimes java.net.SocketTimeoutException manifests as a JSE here?
+                                completedListener.onTaskFailed(LOCAL_FAILURE_CODE, null);
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:323]: this is weird enough we might want to log it?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java [hunk 2]
```diff
@@ -358,10 +365,15 @@ public void getRank(final long pageStart, final long pageEnd,
...
-                                    RankResponse.class);
-                            r.setSelected(selected);
-                            completedListener.onTaskSucceeded(r);
+                            try {
+                                RankResponse r = new Gson().fromJson(responseBody.charStream(),
+                                        RankResponse.class);
+                                r.setSelected(selected);
+                                completedListener.onTaskSucceeded(r);
+                            } catch (JsonSyntaxException e) {
+                                //ALIBI: sometimes java.net.SocketTimeoutException manifests as a JSE here?
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:374]: same

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java [hunk 3]
```diff
@@ -38,6 +39,7 @@
 import java.io.FileOutputStream;
 import java.io.FileReader;
 import java.io.IOException;
+import java.net.SocketTimeoutException;
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:42]: doesn't look like this is used?

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java]: quite right.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:373]: based on stacko's results, I suspect it's pretty standard, really.

Comment (bobzilladev) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/net/WiGLEApiManager.java:373]: ok dokey

**Pull request merged**